pyodide-build/pyodide_build/buildpkg.py

```diff
@@ -451,8 +438,34 @@ def compile(
                 replace_libs=replace_libs,
                 replay_from=replay_from,
             )
-        install_for_distribution()
-    del bash_runner.env["_PYTHON_HOST_PLATFORM"]
+
+
+def set_host_platform(wheel_dir: Path, platform: str):
+    dist_info = next(wheel_dir.glob("*.dist-info"))
+    WHEEL = dist_info / "WHEEL"
+    lines = WHEEL.read_text().splitlines()
+    result_lines = []
+    for line in lines:
+        if line.startswith("Tag:"):
+            line = line.rpartition("-")[0] + "-" + platform
+        result_lines.append(line)
+    WHEEL.write_text("\n".join(result_lines))
+
+
+def update_wheel_platform(build_dir: Path, wheel_dir: Path):
+    import distutils.dir_util
+
+    # for some reason the .so name and the wheel name swap order of architecture
+    # and OS?
+    set_host_platform(wheel_dir, "emscripten_wasm32")
+    pywasmcross.clean_out_native_artifacts(wheel_dir)
+    lib_dir = next(build_dir.glob("lib*"))
+    distutils.dir_util.copy_tree(str(lib_dir), str(wheel_dir))
```

Comment:
Maybe use `shutil.copytree` instead? distutils is deprecated.

Reply:
They don't do the same thing and we need exactly the distutils behavior. We are trying to copy what setuptools does, which is they use this function. But we can remove it with #2256.

Reply:
If you look at the packages CI for [71a07ce](https://github.com/pyodide/pyodide/pull/2238/commits/71a07ce02a55e7c2df8ffb0dcd14fd4039fa0ad7) you can see the problems we get with `shutil.copytree`.

Reply:
Thanks for the explanation!